lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -71,7 +72,13 @@ def initialize(options = {})
       def purchase(money, payment_method, options = {})
         MultiResponse.run do |r|
           r.process { authorize(money, payment_method, options) }
-          r.process { capture(money, r.authorization, options.merge(authorization_validated: true)) } unless options[:skip_capture]
+          unless options[:skip_capture]
+            r.process do
+              response = capture(money, r.authorization, options.merge(authorization_validated: true))
+              response.params['avs_result'] = r.avs_result
+              response
+            end
+          end
         end
```

Comment:
This looks great and unit tests are passing (I haven't been able to get the remote tests yet due to a slow connection 😅 ) I think a test to confirm the `avs_result` from the authorize is included on the response after the capture could be good to add too :)